keps/sig-node/xxx-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,1408 @@
+-->
+
+Dynamic resource allocation introduces a new API for requesting
+arbitrary resources with parameters that can be different depending on
+what kind of resource gets requested. Resource allocations can be
+ephemeral (specified in a Pod spec, allocated and destroyed together
+with the Pod) and persistent (lifecycle managed separately from a Pod
+and used for multiple different Pods).
+
+Resources are managed by plugins that communicate with central
+Kubernetes components, in particular the kube-scheduler, by updating
+objects stored in the kube-apiserver. Communication with kubelet is
+handled through local Unix domain sockets and the plugin registration
+mechanism, using a new plugin type and a new gRPC interface. Towards
+the container runtime, kubelet and those plugins then must use the
+[Container Device Interface
+(CDI)](https://github.com/container-orchestrated-devices/container-device-interface)
+to expose the resources.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Originally, Kubernetes and its scheduler only tracked CPU and RAM as
+resources for containers. Later, support for storage and discrete,
+countable per-node extended resources was added. The device plugin
+interface then made such local resources available to containers. But
+for many newer devices, this approach and the Kubernetes API for
+requesting these custom resources are too limited. This KEP addresses
+the following limitations:
+
+- *Device initialization*: When starting a workload I’d like to have
+  the device reconfigured or reprogrammed by the orchestration. For
+  security reasons workloads should not be able to reconfigure devices
+  directly.
+
+  *Limitation*: Currently, it’s impossible to specify the desired
+  device properties that are required for reconfiguring devices.
+
+- *Device cleanup*: When my workload is finished, I would like to have
+  a mechanism for cleanup of the device, that will ensure that device
+  does not contain traces/parameters/data from previous workloads and
+  appropriate power state/shutdown.
+
+  *Limitation*: Post-stop actions are not supported.
+
+- *Partial allocation*: When deploying a container I’d like to be able
+  to use part of the shareable device inside a container and other
+  containers should be able to use other free resources on the same
+  device.
+
+  *Limitation*: Current implementation of the device plugin doesn’t
+  allow to allocate part of the device because parameters are too limited
+  and Kubernetes doesn't have enough information about the extended
+  resources on a node to decide whether they can be shared.
```

Comment:
I don't know if I agree with this one. The plugin can divide up resources however it wants (assuming the the resources it manages are "divisible") and advertise them in smaller chunks. Maybe you're talking about a more dynamic splitting of resources after the initial set of resources have been advertised to the kubelet?